zdm/iteration.py

```diff
@@ -869,55 +886,62 @@ def cube_likelihoods(grids,surveys,psetmins,psetmaxes,npoints,
     # expected number for best-fit constant
     
     """
+    npoints = np.array([item['n'] for key, item in vparam_dict.items()])
+    vp_keys = list(vparam_dict.keys())
+    state = grids[0].state # Any grid will do
     
     # check feasible range of job number
     if (howmany <= 0) or (run <= 0) or np.prod(np.abs(npoints)) < (run-1)*howmany+1:
-        print("Invalid range of run=",run," and howmany=",howmany," for ",np.prod(npoints)," points")
+        print("Invalid range of run=",run," and howmany=",howmany," for ",np.prod(
+            np.abs(npoints))," points")
         exit()
     
-    NPARAMS=8
-    nsets=len(grids)
-    lls=np.zeros([nsets])
+    NPARAMS = len(vparam_dict)
+    PARAMS = list(vparam_dict.keys())
+    nsets = len(grids)
+    lls = np.zeros([nsets])
     
     f=open(outfile,'a+')
```

Comment:
We probably want it to output a binary file eventually. Or asap. Then we provide a reader which loads the param class so we know the length and the order. Again, a non-essential to-do.

Reply:
agreed.  I might implement it.